.orchestra/ci/ci-hook/app.py

```diff
@@ -132,34 +186,206 @@ def trigger_ci(username, repo_url, ref, before, after):
+            "started": f"The CI job for this merge request has been started for commit {sha}.\n\n"
+                       f"View the status: {ci_job_url}{data['build_id']}",
+            "completed": f"**Success!** The CI job has passed for commit {sha}",
+            "failed": f"**Error!** The CI job has failed for commit {sha}",
+            "cancelled": f"**Cancelled.** The CI job has been cancelled for commit {sha}",
+            "skipped": f"**Skipped.** The CI job was skipped for commit {sha}"
+        }
+        message = messages.get(LAB_TO_LAB_STATUS_MAP[data["build_status"]])
+
+        if message:
+            with impersonate(admin_gl, ci_user) as token:
+                user_gl = gitlab.Gitlab(GITLAB_URL, private_token=token)
+                user_gl.auth()
+
+                mr_notes = user_gl.projects.get(metadata['gitlab_project_id']).mergerequests.get(
+                    metadata['gitlab_mr_iid']).notes
+                mr_notes.create({
+                    "body": message,
+                    "merge_request_diff_sha": metadata['head_sha']
+                })
+    else:
+        return "Invalid platform ID\n", 500
+
+    return "All good\n", 200
+
+
 @app.route('/ci-hook/gitlab', methods=["POST"])
 def gitlab_hook():
     headers = dict(request.headers)
     if headers.get("X-Gitlab-Token", "") != GITLAB_SECRET:
         return "Invalid token\n", 403
 
+    event = headers.get("X-Gitlab-Event", "")
     data = request.json
-    trigger_ci(data["user_username"],
-               data["project"]["git_http_url"] + " " + data["project"]["git_ssh_url"],
-               data["ref"],
-               data["before"],
-               data["after"])
-    return "All good\n"
+    attributes = data.get("object_attributes", {})
+
+    if event == "Push Hook":
+        trigger_ci(
+            data["user_username"],
+            data["project"]["git_http_url"] + " " + data["project"]["git_ssh_url"],
+            data["ref"]
+        )
+
+    elif event == "Merge Request Hook" and attributes.get("action", "") in ("open", "update") \
+            and attributes.get("state", "") == "opened":
+        trigger_ci(
+            data["user"]["username"],
+            attributes["source"]["git_http_url"] + " " + attributes["source"]["git_ssh_url"],
+            attributes["source_branch"],
+            {
+                "platform": "gitlab",
+                "gitlab_project_id": data['project']['id'],
+                "gitlab_mr_iid": attributes['iid'],
+                "head_sha": attributes['last_commit']['id']
+            }
+        )
+
+    return "All good\n", 200
 
 
 @app.route('/ci-hook/github', methods=["POST"])
 def github_hook():
     headers = dict(request.headers)
-    signature = 'sha256=' + hmac.new(GITHUB_SECRET, request.data, hashlib.sha256).hexdigest()
+
+    event = headers.get("X-Github-Event", "")
+    if event in ("check_suite", "check_run"):
+        secret = GITHUB_APP_SECRET
+    else:
+        secret = GITHUB_SECRET
```

Comment:
A bit sad. Do you envision an opportunity to unify these?

Reply:
Yes, let's do this.

Reply:
Done (old webhooks are still configured but they aren't currently being accepted by the deployed hook)

Reply:
Should I remove all the webhooks from all repositories?

Reply:
On GH, yes.

Reply:
Done, I've also removed the webhook for revng-c on gitlab.

Reply:
> Done, I've also removed the webhook for revng-c on gitlab.

Do you no longer want it to trigger the CI? On GitLab it is still needed, you need "push" and "merge request" for all repos.

Reply:
Ah shit. Can you set it up again? `:P`

Reply:
Done